src/llama_stack_client/lib/agents/client_tool.py

```diff
@@ -64,3 +65,105 @@ def run(
         self, messages: List[Union[UserMessage, ToolResponseMessage]]
     ) -> List[Union[UserMessage, ToolResponseMessage]]:
         raise NotImplementedError
+
+
+class SingleMessageClientTool(ClientTool):
+    """
+    Helper class to handle custom tools that take a single message
+    Extending this class and implementing the `run_impl` method will
+    allow for the tool be called by the model and the necessary plumbing.
+    """
+
+    def run(self, messages: List[Union[UserMessage, ToolResponseMessage]]) -> List[ToolResponseMessage]:
+        assert len(messages) == 1, "Expected single message"
+
+        message = messages[0]
+
+        tool_call = message.tool_calls[0]
+
+        try:
+            response = self.run_impl(**tool_call.arguments)
+            response_str = json.dumps(response, ensure_ascii=False)
+        except Exception as e:
+            response_str = f"Error when running tool: {e}"
+
+        message = ToolResponseMessage(
+            call_id=tool_call.call_id,
+            tool_name=tool_call.tool_name,
+            content=response_str,
+            role="tool",
+        )
+        return [message]
+
+    @abstractmethod
+    def run_impl(self, *args, **kwargs):
+        raise NotImplementedError()
+
+
+T = TypeVar("T", bound=Callable)
+
+
+def tool(func: T) -> ClientTool:
+    """
+    Decorator to convert a function into a ClientTool.
+    Usage:
+        @tool
+        def add(x: int, y: int) -> int:
+            '''Add 2 integer numbers
+
+            :param x: integer 1
+            :param y: integer 2
+            :returns: sum of x + y
+            '''
+            return x + y
+    """
+
+    class WrappedTool(SingleMessageClientTool):
```

Comment:
nit: prefix with underscore since this should be an internal use class

Reply:
done